### ISSUE TITLE
[GBA] Savedata doesn't persist in The Legend of Zelda: The Minish Cap

### ISSUE THREAD

**Issue by OrdinaryCarrot16** [higan-emu/ares#687]
**Issue**

I was doing a play-through of The Minish Cap, and stopped playing shortly after beating the first temple and saved the game. It seemingly went fine inside the game, but the next day when I went to play it, my save file was missing. I looked inside the saves folder and saw an 'eeprom' file relating to the game, but it was 0 KB. Luckily I made a save state before quitting the game, but those aren't future proof. Future attempts to save the game (after loading the save state) are still unsuccessful.

```
file: The Legend of Zelda - The Minish Cap (USA).gba
sha256: bedc74df62755f705398273de8ed3bc59be610cf55760d0b9aa277f1f5035e73
```
version: ares v129

**Comment by LukeUsher** [comment]
Fixed in d5c094e9e8bdc381d20b7a7d268bb3a5f98ea2a3

**Comment by LukeUsher** [comment]
Reopened this and #717,

The gba eeprom emulation needs a larger rework in order to fix one without breaking the other.

**Comment by LukeUsher** [comment]
fixed in 972495df89353a73bee91a3fe917853f8733d515
